File: uniseek_vue/src/api/task.ts
import request from './index'
import type { ApiResponse } from './auth'

// 分页结果结构
export interface PageResult<T> {
  records: T[]
  total: number
  page: number
  pageSize: number
  totalPages: number
}

// 职位搜索参数
export interface TaskSearchParams {
  keyword?: string
  categoryId?: number
  regionId?: number
  jobType?: number
  salaryMin?: number
  salaryMax?: number
  salaryUnit?: number
  address?: string
  tag?: string
  sortBy?: string
  sortOrder?: string
  page?: number
  pageSize?: number
}

// 职位视图（匹配后端 TaskVO）
export interface TaskVO {
  id: number
  enterpriseId: number
  categoryId: number
  regionId: number
  title: string
  description: string
  salaryMin: number
  salaryMax: number
  salaryUnit: number
  jobType: number
  totalQuota: number
  remainingQuota: number
  address: string
  tag: string[]
  longitude: number
  latitude: number
  status: number
  deadline: string | null
  createTime: string
  updateTime: string
  enterpriseName: string
  categoryName: string
  regionName: string
  hasApplied: boolean
  applicationCount: number
}

// 发布职位请求参数
export interface CreateTaskParams {
  categoryId: number
  regionId: number
  title: string
  description: string
  salaryMin: number
  salaryMax: number
  salaryUnit: number
  jobType: number
  totalQuota: number
  address: string
  tag: string[]
  deadline: string | null
}

/** GET /api/tasks 分页搜索职位 */
export const searchTasks = async (params: TaskSearchParams) => {
  const res = await request.get<ApiResponse<PageResult<TaskVO>>>('/tasks', { params })
  return res.data
}

/** GET /api/tasks/:id 获取职位详情 */
export const getTaskById = async (id: number) => {
  const res = await request.get<ApiResponse<TaskVO>>(`/tasks/${id}`)
  return res.data
}

/** POST /api/tasks 发布职位 */
export const createTask = async (params: CreateTaskParams) => {
  const res = await request.post<ApiResponse<TaskVO>>('/tasks', params)
  return res.data
}

/** PUT /api/tasks/:id 更新职位 */
export const updateTask = async (id: number, params: CreateTaskParams) => {
  const res = await request.put<ApiResponse<TaskVO>>(`/tasks/${id}`, params)
  return res.data
}

/** PUT /api/tasks/:id/status 修改职位状态 */
export const updateTaskStatus = async (id: number, targetStatus: number) => {
  const res = await request.put<ApiResponse<void>>(`/tasks/${id}/status?targetStatus=${targetStatus}`)
  return res.data
}

/** GET /api/enterprise/tasks 查询本企业职位列表 */
export const getEnterpriseTasks = async (page = 1, pageSize = 100) => {
  const res = await request.get<ApiResponse<PageResult<TaskVO>>>(`/enterprise/tasks?page=${page}&pageSize=${pageSize}`)
  return res.data
}

File: uniseek_java/src/main/java/com/uniseek/entity/Task.java
package com.uniseek.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.Version;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 职位任务实体
 */
@Data
@TableName("task")
public class Task {

    /** 职位 ID */
    @TableId
    private Long id;

    /** 所属企业 ID */
    private Long enterpriseId;

    /** 职位分类 ID */
    private Long categoryId;

    /** 地区 ID */
    private Long regionId;

    /** 职位标题 */
    private String title;

    /** 职位描述 */
    private String description;

    /** 最低薪资（元） */
    private BigDecimal salaryMin;

    /** 最高薪资（元） */
    private BigDecimal salaryMax;

    /** 薪资单位：0 日结 / 1 时薪 / 2 月结 */
    private Integer salaryUnit;

    /** 工作类型：1 全职 / 2 兼职 / 3 实习 */
    private Integer jobType;

    /** 招聘总名额 */
    private Integer totalQuota;

    /** 剩余名额 */
    private Integer remainingQuota;

    /** 工作地址 */
    private String address;

    /** 发布企业名称（非数据库字段，关联查询填充） */
    @TableField(exist = false)
    private String enterpriseName;

    /** 职位标签，JSON数组格式 */
    @TableField("tag")
    private String tag;

    /** 经度 */
    private BigDecimal longitude;

    /** 纬度 */
    private BigDecimal latitude;

    /** 状态：0 待审 / 1 招聘中 / 2 已满员 / 3 已过期 / 4 已下架 */
    private Integer status;

    /** 驳回原因 */
    private String rejectReason;

    /** 乐观锁版本号 */
    @Version
    private Integer version;

    /** 报名截止时间 */
    private LocalDateTime deadline;

    /** 审核通过时间 */
    @TableField("audit_time")
    private LocalDateTime auditTime;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}
